File: backend/internal/domain/transport.go
package domain

import "time"

type TransportListing struct {
	ID               int64      `json:"id"`
	ProviderID       int64      `json:"provider_id"`
	Kind             string     `json:"kind"`
	CompanyName      string     `json:"company_name"`
	DriverNames      string     `json:"driver_names"`
	VehicleBrand     string     `json:"vehicle_brand"`
	VehiclePhotoURL  string     `json:"vehicle_photo_url,omitempty"`
	Phone            string     `json:"phone,omitempty"`
	Email            string     `json:"email,omitempty"`
	Telegram         string     `json:"telegram,omitempty"`
	Whatsapp         string     `json:"whatsapp,omitempty"`
	Viber            string     `json:"viber,omitempty"`
	PriceAmount      float64    `json:"price_amount"`
	PriceCurrency    string     `json:"price_currency"`
	SeatsTotal       int        `json:"seats_total"`
	ParcelsAccepted  bool       `json:"parcels_accepted"`
	ParcelsTerms     string     `json:"parcels_terms,omitempty"`
	DepartTime       string     `json:"depart_time"`
	ArriveTimeApprox string     `json:"arrive_time_approx,omitempty"`
	Status           string     `json:"status"`
	CreatedAt        time.Time  `json:"created_at,omitempty"`
	UpdatedAt        time.Time  `json:"updated_at,omitempty"`
	Stops            []TransportStop      `json:"stops,omitempty"`
	Departures       []TransportDeparture `json:"departures,omitempty"`
	ProviderName     string     `json:"provider_name,omitempty"`
	ProviderSlug     string     `json:"provider_slug,omitempty"`
	ContactsUnlocked bool       `json:"contacts_unlocked,omitempty"`
	CarrierType      string     `json:"carrier_type,omitempty"`
	VerifiedUkrainian bool      `json:"verified_ukrainian,omitempty"`
	SubscriptionActive bool     `json:"subscription_active,omitempty"`
	Description        string     `json:"description,omitempty"`
	BookingsCount      int        `json:"bookings_count,omitempty"`
	SeatsAvailable     *int       `json:"seats_available,omitempty"`
}

type TransportStop struct {
	ID         int64  `json:"id,omitempty"`
	ListingID  int64  `json:"listing_id,omitempty"`
	CityID     int64  `json:"city_id"`
	CityName   string `json:"city_name,omitempty"`
	CitySlug   string `json:"city_slug,omitempty"`
	CountryName string `json:"country_name,omitempty"`
	SortOrder  int    `json:"sort_order"`
}

type TransportDeparture struct {
	ID        int64  `json:"id,omitempty"`
	ListingID int64  `json:"listing_id,omitempty"`
	DepartOn  string `json:"depart_on"`
	ArriveOn  string `json:"arrive_on,omitempty"`
	SeatsLeft *int   `json:"seats_left,omitempty"`
}

File: packages/api-client/src/api/types/transport.ts
export type TransportStop = {
  id?: number
  city_id: number
  city_name?: string
  city_slug?: string
  country_name?: string
  sort_order: number
}

export type TransportDeparture = {
  id?: number
  depart_on: string
  arrive_on?: string
  seats_left?: number | null
}

export type TransportListing = {
  id: number
  provider_id: number
  kind: 'regular' | 'occasional'
  company_name: string
  driver_names: string
  vehicle_brand: string
  vehicle_photo_url?: string
  phone?: string
  email?: string
  telegram?: string
  whatsapp?: string
  viber?: string
  price_amount: number
  price_currency: string
  seats_total: number
  parcels_accepted: boolean
  parcels_terms?: string
  depart_time: string
  arrive_time_approx?: string
  status: string
  provider_name?: string
  provider_slug?: string
  contacts_unlocked: boolean
  subscription_active?: boolean
  logged_in?: boolean
  carrier_type?: string
  verified_ukrainian?: boolean
  description?: string
  seats_available?: number
  bookings_count?: number
  stops: TransportStop[]
  departures: TransportDeparture[]
}

export type TransportListingInput = {
  kind: 'regular' | 'occasional'
  company_name?: string
  driver_names: string
  vehicle_brand: string
  vehicle_photo_url?: string
  phone?: string
  email?: string
  telegram?: string
  whatsapp?: string
  viber?: string
  price_amount: number
  price_currency?: string
  seats_total: number
  parcels_accepted: boolean
  parcels_terms?: string
  depart_time: string
  arrive_time_approx?: string
  status?: string
  stops: Array<{ city_id: number; sort_order: number }>
  departures: TransportDeparture[]
  provider_slug?: string
  provider_name?: string
}
